File: vm/tests/all/interop/simpleClass/simpleClassPoint.cpp
//simpleClass.cpp
#include<iostream.h>
#include<polyglot.h>
#include<math.h>

class Point {
	private:
	int x;
	int y;
	public:
	int getX();
	int getY();
	void setX(int dX);
	void setY(int dY);
	double distanceTo(Point* other);
	void printMe();
};

POLYGLOT_DECLARE_CLASS(Point);

int Point::getX() {
	return x;
}

int Point::getY() {
	return y;
}

void Point::setX(int dX) {
	x=dX;
}

void Point::setY(int dY) {
	y=dY;
}

void Point::printMe() {
	cout << "Point: x=" << x << ", y=" << endl;
}

double distanceTo(Point* other) {
	double dX = x - other->x;
	double dY = y - other->y;
	return sqrt(dX*dX+dY*dY);
}

File: vm/tests/all/interop/simpleClass/simpleClassPoint.py
#simpleClass.py
import math

class Point:
	def __init__(self, x, y):
		self.x=x
		self.y=y
	
	def getX(self):
		return self.x
	
	def getY(self):
		return self.y
	
	def setX(self, val):
		self.x=val

	def setY(self, val):
		self.y=val

	def printMe(self):
		print("Point: x="+str(self.x)+", y="+str(self.y))
	
	def distanceTo(self, other) {
		dX = self.x - other.x
		dY = self.y - other.y
		return math.sqrt(dX*dX+dY*dY)
	}


